llvm/lib/Target/AArch64/AArch64SchedNeoverseN2.td

```diff
@@ -882,7 +906,7 @@ def : SchedAlias<WriteFImm, N2Write_2c_1V>;
 def : InstRW<[N2Write_2c_1V], (instrs FMOVHr, FMOVSr, FMOVDr)>;
 
 // FP transfer, from gen to low half of vec reg
-def : InstRW<[N2Write_3c_1M0], (instrs FMOVWHr, FMOVXHr, FMOVWSr, FMOVXDr,
+def : InstRW<[N2Write_0or3c_1M0], (instrs FMOVWHr, FMOVXHr, FMOVWSr, FMOVXDr,
                                         FMOVHWr, FMOVHXr, FMOVSWr, FMOVDXr)>;
```

Comment:
This is unrelated to your patch, but since we're here: based on the SWOG, this doesn't look right for FP to GP moves?

Reply:
Thanks, based on the SWOG it looks like the throughput may now be slightly higher than advertised? The new change still looks better than before, so up to you if you want to follow up with a fix.